denonavr/denonavr.py

```diff
@@ -1032,6 +1049,14 @@ def _get_status_from_xml_tags(self, root, relevant_tags):
             elif child.tag == "FriendlyName" and self._name is None:
                 self._name = child[0].text
                 relevant_tags.pop(child.tag, None)
+            elif (child.tag == "selectSurround" or child.tag == "SurrMode"):
+                sound_mode_raw = child[0].text.rstrip()
+                self._sound_mode = self.match_sound_mode(sound_mode_raw)
+                self._sound_mode_raw = sound_mode_raw
+                if "selectSurround" in relevant_tags.keys():
+                    relevant_tags.pop("selectSurround", None)
```

Comment:
Could you pop both keys from the list, when the first one was found?
They seem to cover the same information. At the moment there might be a second HTTP request, which is not needed, because the STATUS_URL is called first and there is a second call to MAINZONE_URL if some tags are not found yet.

Reply:
I am poping both keys when the first one is found.
when one of the keys is found `elif (child.tag == "selectSurround" or child.tag == "SurrMode"):` , both keys are poped using `relevant_tags.pop("selectSurround", None)` and `relevant_tags.pop("SurrMode", None)`.
The `if "selectSurround" in relevant_tags.keys():` and `if "SurrMode" in relevant_tags.keys():` will in principle always return true, however I put those in to prevent the occasion that later when the code is changed there is no problem with trying to remove a key that does not exist. In the case that `if "selectSurround" in relevant_tags.keys():` or `if "SurrMode" in relevant_tags.keys():` returns false, the key is not in the list anyway so their is no need to pop it.

Reply:
Therefore I think this part of the code is correct, and a second HTTP request is not possible unless none of the two keys ("selectSurround" and "SurrMode") are found.

Reply:
I am poping both keys when the first one is found.
when one of the keys is found elif (child.tag == "selectSurround" or child.tag == "SurrMode"):, both keys are poped using relevant_tags.pop("selectSurround", None) and relevant_tags.pop("SurrMode", None).
The if "selectSurround" in relevant_tags.keys(): and if "SurrMode" in relevant_tags.keys(): will in principle always return true, however I put those in to prevent the occasion that later when the code is changed there is no problem with trying to remove a key that does not exist. In the case that if "selectSurround" in relevant_tags.keys(): or if "SurrMode" in relevant_tags.keys(): returns false, the key is not in the list anyway so their is no need to pop it.

Reply:
Therefore I think this part of the code is correct, and a second HTTP request is not possible unless none of the two keys ("selectSurround" and "SurrMode") are found.

Reply:
ah ok, I was confused by the if statement.
Btw. you don't need the if statements here. If you `pop` with an default value, there won't be an exception even if the key is not in the list